packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -258,12 +258,28 @@ export class ProjectDataSource {
     this.ctx.coreData.app.relaunchBrowser = relaunchBrowser
   }
 
-  async specPatterns (): Promise<{
+  async specPatterns (testingType?: TestingType): Promise<{
     specPattern?: string[]
     excludeSpecPattern?: string[]
   }> {
     const toArray = (val?: string | string[]) => val ? typeof val === 'string' ? [val] : val : undefined
 
+    const configFile = await this.ctx.lifecycleManager.getConfigFileContents()
+
+    if (testingType === 'e2e') {
+      return {
+        specPattern: toArray(configFile.e2e?.specPattern ?? defaultSpecPattern.e2e),
+        excludeSpecPattern: toArray(configFile.e2e?.excludeSpecPattern ?? defaultExcludeSpecPattern.e2e),
+      }
+    }
+
+    if (testingType === 'component') {
+      return {
+        specPattern: toArray(configFile.component?.specPattern ?? defaultSpecPattern.component),
+        excludeSpecPattern: toArray(configFile.component?.excludeSpecPattern ?? defaultExcludeSpecPattern.component),
+      }
+    }
+
     const config = await this.getConfig()
 
```

Comment:
I think this logic should get pulled out into it's own function.  Right now, there are two totally different paths depending on if `testintType` gets passed in.  This conditional logic could at the least get pushed down to the `matchesSpecPattern` function below.  At the very least, I think some good comments would be good to explain the difference between what the original functions are doing and what the new versions are trying to accomplish. 

I could not find any existing unit tests for these functions, but having a few that could further demonstrate the functionality would be helpful as well.

Reply:
Split this out into new function and added tests